### DISCUSSION TITLE
Detect Arrow Headshot

### DISCUSSION THREAD

**Discussion by para-dise** [Minestom/Minestom#1229] [Help] [answered]
Hey, is there any way to detect if an arrow headshots a player? I was reading https://www.spigotmc.org/threads/headshot-arrow-damage.419479/ and here they did `y - shotY > 1.35d;` however if I do
```java
event.getEntity().getPosition().y();
event.getHitEntity().getPosition().y();
```

I always get the same Y coordinate for both the arrow and the hit entity.

**Comment by DevBlokJochem** [comment]
event.getEntity = arrow
event.getHitEntity = target

You should calculate the distance between the eye position of the entity and where the arrow hit the target. If it is smaller than 0.2 (or something), you did a headshot.

**Comment by MrlnHi** [comment, marked as answer]
Now I haven't tested this, but I'm assuming the y coordinate of the arrow gets lost when it hits an entity or something. If that is the case, you can try setting a Tag on the arrow every tick, that saves its y coordinate of the previous tick. Then you can use that one
